src/Illuminate/Notifications/HasDatabaseNotifications.php

```diff
@@ -33,4 +35,22 @@ public function unreadNotifications()
     {
         return $this->notifications()->whereNull('read_at');
     }
+
+    /**
+     * Boot the trait on the notifiable model.
+     *
+     * @return void
+     */
+    public static function bootHasDatabaseNotifications()
+    {
+        static::deleting(function ($notifiable) {
+            if (in_array(SoftDeletes::class, class_uses_recursive($notifiable))) {
+                if (! $notifiable->forceDeleting) {
```

Comment:
@ahmed-aliraqi  To reduce code indentation I think you may better move this condition to the above one. Also make sure to add a test for this.